### DISCUSSION TITLE
For unsupported languages, is there a way to use this implementation by feeding it the AST instead of the files?

### DISCUSSION THREAD

**Discussion by mahtab-nejati** [GumTreeDiff/gumtree#295] [Q&A]
I am working on a project on CMake and would like to export the edit script of the `.cmake` files in my codebase. Is there a way I could use this package by feeding it the two ASTs of the code?

**Comment by jrfaller** [comment]
Hi!

I am not sure to understand what you want?

Cheers.

**Reply by mahtab-nejati** [reply]
Specifically, my question is how can I extend this tool to support diffing of other programming languages, such as CMake?

**Comment by mahtab-nejati** [comment]
Specifically, my question is how can I extend this tool to support diffing of other programming languages, such as CMake?

**Reply by jrfaller** [reply]
Ok that's much clearer thanks! Do you know a CMake parser?

**Reply by mahtab-nejati** [reply]
The CMake parser I am using is in python ([PyPi page](https://pypi.org/project/cmakeast/) and [GitHub repo](https://github.com/polysquare/cmake-ast). I was wondering if I could use the output AST and somehow plug it into the GumTree tool for diffing.

**Reply by jrfaller** [reply]
Hi!

In this case this might be fairly easy. You only need to build a xml convertor to gumtree external format. There is some explanation on how to do this here : https://github.com/GumTreeDiff/gumtree/issues/164

Cheers!

PS: see also this : https://github.com/uyha/tree-sitter-cmake it could be easier since we have a built-in support for tree-sitter parsers.

**Comment by mahtab-nejati** [comment]
Hi,

I did manage to make the tree-sitter-cmake work. I also went through issue #164 which you referred me to.

From what I understand, I need to:

1. Provide support for CMake by adding the file CMakeTreeSitterTreeGenerator.java to [this directory](https://github.com/GumTreeDiff/gumtree/tree/main/gen.treesitter/src/main/java/com/github/gumtreediff/gen/treesitter/)
2. Use [this repository](https://github.com/GumTreeDiff/tree-sitter-parser) and integrate [tree-sitter-cmake](https://github.com/uyha/tree-sitter-cmake) into this.

Is that all?

**Reply by jrfaller** [reply]
Hi!

Yes it's all. Normally it's better to perform 2. before 1. this way I can refresh the docker image so that the right version of tree-sitter-parser is available for GumTree.

Cheers.
